emulators/qemu-guest-agent/src/etc/inc/plugins.inc.d/qemuguestagent.inc

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ *    Copyright (C) 2021 Frank Wall
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+function qemuguestagent_enabled()
+{
+    global $config;
+
+    return isset($config['OPNsense']['QemuGuestAgent']['general']['Enabled']) &&
+    $config['OPNsense']['QemuGuestAgent']['general']['Enabled'] == 1;
+}
+
+/**
+ *  register legacy service
+ * @return array
+ */
+function qemuguestagent_services()
+{
+    $services = array();
+
+    if (!qemuguestagent_enabled()) {
+        return $services;
+    }
+
+    $services[] = array(
+        'description' => gettext('QEMU Guest Agent'),
+        'pidfile' => '/var/run/qemu-ga.pid',
+        'configd' => array(
+            'restart' => array('qemuguestagent restart'),
+            'start' => array('qemuguestagent start'),
+            'stop' => array('qemuguestagent stop'),
+        ),
+        'name' => 'qemuguestagent',
```

Comment:
Normally this is the binary‘s name

Reply:
Ah, I didn't know that. Will rename it to `qemu-ga`then.

Reply:
hm, if I rename it to `qemu-ga`, then it does not show up in `System: Diagnostics: Services` anymore.

Reply:
I‘m not aware of such limitations. Syntax error maybe? It doesn’t read the file on syntax error.

Reply:
Silly me. I've tested with a brand new config, so `qemuguestagent_enabled()` returned `false` and as a result the service did not show up in the GUI :)